### ISSUE TITLE
Empty response when bad oauth2 credentials supplied (e.g. expired access token)

### ISSUE THREAD

**Issue by dorongutman** [zfcampus/zf-mvc-auth#99]
In the authenticate method of the oauth2 adapter (https://github.com/zfcampus/zf-mvc-auth/blob/master/src/Authentication/OAuth2Adapter.php#L135) - if the response is an IdentityInterface, the output is a json that looks like this (due to the SendApiProblemResponseListener, which parses the ApiProblem response which is generated later on):

```json
{
  "type": "http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html",
  "title": "Forbidden",
  "status": 403,
  "detail": "Forbidden"
}
```

However - in the case of an error from the response of the oauth2 server (https://github.com/zfcampus/zf-mvc-auth/blob/master/src/Authentication/OAuth2Adapter.php#L154-L156), what returns is a Zend\Http\Response, causing the response to be an empty response with only the 401 status code but without a body (https://github.com/zfcampus/zf-mvc-auth/blob/master/src/Authentication/DefaultAuthenticationListener.php#L189-L191).

This issue started happening to me after I've updated apigility from 1.0.* to the latest.

**Comment by developer-devPHP** [comment]
I have a same problem. If, for example, the access token is invalid, I got an empty body, but in headers have this  `` WWW-Authenticate:Bearer realm="Service", error="invalid_token", error_description="The access token provided is invalid" ``
![2015-09-09_1948](https://cloud.githubusercontent.com/assets/3036510/9766916/8a3df050-572c-11e5-80f0-bfd8d21afcac.png)

**Comment by TomHAnderson** [comment]
Your response isn't empty, it's 401 Unauthorized

**Comment by developer-devPHP** [comment]
Yes it is 401, but as you can see the response in header and not in body. I mean it not in apiproblem syntax.

**Comment by developer-devPHP** [comment]
But I think it should be something like this 
```json
{
  "type": "http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html",
  "title": "Unauthorized",
  "status": 401,
  "detail": "Unauthorized"
} 
```

**Comment by TomHAnderson** [comment]
10.4.2 does not suggest any body in the response to a 401:
http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

**Comment by nuxwin** [comment]
@TomHAnderson

You're assertion is perfectly valid but since apigility cast response to ApiProblemResponse in case of response denoting a problem (which include a jSON body) we should be consistent.

See also: #108

**Comment by webaaz** [comment]
I agree with that. I need consistency on JSON response

**Comment by weierophinney** [comment]
This repository has been closed and moved to laminas-api-tools/api-tools-mvc-auth; a new issue has been opened at https://github.com/laminas-api-tools/api-tools-mvc-auth/issues/13.
